plugin-server/src/cdp/hog-transformations/hog-transformer.service.ts

```diff
@@ -72,47 +85,134 @@ export class HogTransformerService {
     public async start(): Promise<void> {
         const hogTypes: HogFunctionTypeType[] = ['transformation']
         await this.hogFunctionManager.start(hogTypes)
+        this.kafkaProducer = await KafkaProducerWrapper.create(this.hub)
+    }
+
+    private produceAppMetric(metric: HogFunctionAppMetric): Promise<void> {
+        const appMetric: AppMetric2Type = {
+            app_source: 'hog_function',
+            ...metric,
+            timestamp: castTimestampOrNow(null, TimestampFormat.ClickHouse),
+        }
+
+        return this.kafkaProducer
+            .queueMessages([
+                {
+                    topic: KAFKA_APP_METRICS_2,
+                    messages: [
+                        {
+                            value: safeClickhouseString(JSON.stringify(appMetric)),
+                            key: appMetric.app_source_id,
+                        },
+                    ],
+                },
+            ])
+            .catch((error) => {
+                status.error('⚠️', `failed to produce app metric: ${error}`)
+            })
     }
 
-    public transformEvent(event: PluginEvent): Promise<PluginEvent> {
+    private produceLogs(result: HogFunctionInvocationResult): Promise<void> {
+        const logs = fixLogDeduplication(
+            result.logs.map((logEntry) => ({
+                ...logEntry,
+                team_id: result.invocation.hogFunction.team_id,
+                log_source: 'hog_function',
+                log_source_id: result.invocation.hogFunction.id,
+                instance_id: result.invocation.id,
+            }))
+        )
+
+        return this.kafkaProducer
+            .queueMessages(
+                logs.map((logEntry) => ({
+                    topic: KAFKA_LOG_ENTRIES,
+                    messages: [
+                        {
+                            value: safeClickhouseString(JSON.stringify(logEntry)),
+                            key: logEntry.instance_id,
+                        },
+                    ],
+                }))
+            )
+            .catch((error) => {
+                status.error('⚠️', `failed to produce logs: ${error}`)
+            })
+    }
+
+    private processInvocationResult(result: HogFunctionInvocationResult): Promise<void>[] {
```

Comment:
This feels better. Eventually we can abstract all this into helper methods across the various hog-related use cases